python/arcticdb/storage_fixtures/api.py

```diff
@@ -0,0 +1,182 @@
+        ``create_test_cfg``"""
+        return functools.partial(
+            self._factory_impl, self.libs_from_factory, self.create_test_cfg, default_lib_name, **defaults
+        )
+
+    def slow_cleanup(self, failure_consequence=""):
+        for lib in self.libs_from_factory.values():
+            with handle_cleanup_exception(self, lib, consequence=failure_consequence):
+                lib.version_store.clear()
+        self.libs_from_factory.clear()
+
+        arctic = self.create_arctic()
+        for lib in self.libs_names_from_arctic[:]:
+            with handle_cleanup_exception(self, lib, consequence=failure_consequence):
+                arctic.delete_library(lib)
+
+    def set_permission(self, *, read: bool, write: bool):
+        """Makes the connection to the storage have the given permissions. If unsupported, call ``pytest.skip()``.
+        See ``set_enforcing_permissions`` below."""
+        raise NotImplementedError(type(self).__name__ + " does not implement set_permission()")
+
+    def iter_underlying_object_names(self):
+        raise NotImplementedError(type(self).__name__ + " does not implement iter_underlying_object_names()")
+
+    def copy_underlying_objects_to(self, destination: "StorageFixture"):
+        """Where implemented, copies all objects stored in the backing Storage to the destination.
+        Unless otherwise documented, the ``destination`` argument should be the same type as ``self``."""
+        raise NotImplementedError(type(self).__name__ + " does not implement copy_underlying_objects_to()")
+
+    @classmethod
+    def replace_uri_field(cls, uri: str, field: ArcticUriFields, replacement: str, start=2, end=2):
+        """start & end are these regex groups: 1=field name, 2=field value, 3=optional field separator"""
+        regex = cls._FIELD_REGEX[field]
+        match = regex.search(uri)
+        assert match, f"{uri} does not have {field}"
+        return f"{uri[:match.start(start)]}{replacement}{uri[match.end(end):]}"
+
+
+class StorageFixtureFactory(_SaferContextManager):
+    """For ``StorageFixture``s backed by shared/expensive resources, implement this class to manage those."""
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """Properly clean up the fixture. This should be safe to be called multiple times."""
+
+    @property
+    def enforcing_permissions(self):
+        """Indicates whether this Factory will create Fixtures that enforces permissions.
+
+        If implemented, the set value should affect subsequent calls to ``create_fixture()`` and, if explicitly
+        documented, existing fixtures. The base implementation is hard-wired to ``False``."""
+        return False
+
+    @contextmanager
+    def enforcing_permissions_context(self, set_to=True):
+        saved = self.enforcing_permissions
+        try:
+            self.enforcing_permissions = set_to
```

Comment:
You need a setter for `enforcing_permissions` don't you? It's read-only with the current code.

Reply:
It is meant to be read-only in the base class. Sub-types will supply a setter if they support permissions.